### PR TITLE
add tests with mocked database

### LINKED ISSUE #13: Add tests
👀 https://technology.cloverhealth.com/better-postgresql-testing-with-python-announcing-pytest-pgsql-and-pgmock-d0c569d0602a

### PR DESCRIPTION
**Pull request by geospatial-jeff**
Closes #13

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1,19 +1,46 @@
...
+    because pytest-asyncio event loop is a function scoped fixture and cannot be called within the current scope.  Yields
+    a database url which we pass to our application through a monkeypatched environment variable.
+    """
+    db_url = f"postgresql://{postgresql_my_proc.user}:{postgresql_my_proc.password}@{postgresql_my_proc.host}:{postgresql_port}"
+    conn = psycopg2.connect(dsn=db_url)
+    cursor = conn.cursor()
+    cursor.execute("CREATE EXTENSION postgis")
+    cursor.execute(
+        open(
+            "/Users/jeffalbrecht/Documents/Personal/timvt/Dockerfiles/db/landsat_wrs.sql"
```
Comment (geospatial-jeff) [inline tests/conftest.py:31]: :0

File: tests/conftest.py [hunk 2]
```diff
@@ -1,19 +1,46 @@
...
-    monkeypatch.setenv("POSTGRES_PASS", "rde")
-    monkeypatch.setenv("POSTGRES_DBNAME", "lobster")
-    monkeypatch.setenv("POSTGRES_HOST", "1.2.3.4")
-    monkeypatch.setenv("POSTGRES_PORT", "1234")
+    monkeypatch.setenv("DATABASE_URL", database_url)
 
     from timvt.app import app
 
-    return TestClient(app)
+    with TestClient(app) as app:
```
Comment (vincentsarago) [inline tests/conftest.py:45]: could you make this a fixture?

File: tests/routes/test_tms.py [hunk 3]
```diff
@@ -1,4 +1,25 @@
 """test TileMatrixSets endpoints."""
+from urllib.parse import urlsplit
+
+import mapbox_vector_tile
+
+
+def test_index(app):
```
Comment (vincentsarago) [inline tests/routes/test_tms.py:7]: let's move this in a separate file!

**Comment by vincentsarago** [comment]
🎉 woot the pytest plugin seems great ❤️

**Comment by geospatial-jeff** [comment]
lgtm!

**Review by vincentsarago — APPROVED** [review]

**Pull request merged**